### PR TITLE
Model version update (v2)

### PR DESCRIPTION
**Pull request by DavidVista**
At this point, we have patched the model in the following ways:

- Removed data leakage in the generation of train and test samples from the shared dataset of stems.
- Changed loss function from MSE to a composite one (L1 loss for reconstruction of noise and target, SNR loss for the target, and spectral loss for the target.
- Removed randomized energy level setup for the samples. Instead, added normalization of signal to the range [-1, 1] as proposed in Wave-UNET architecture.
- Changed crop of the model output to the linear interpolation as proposed in the Wave-UNET paper.
- Tuned parameters to avoid overfitting.

Additionally, we worked out the  project structure and split the original notebook into essential ML pipeline stages for further deployment. Moreover, the backend functionality was tested out, and we are ready for the frontend.

Now, the repository contains the v2 model in the `models/trained` folder and the output for the quality test file in the respective folder.

### CODE REVIEW

**Review by Spectre113 — APPROVED** [review]
Outstanding—tight code, accurate outputs, and excellent visuals
